File: project/app/src/main/java/org/owntracks/android/injection/modules/FragmentModules/VersionFragmentModule.java
package org.owntracks.android.injection.modules.FragmentModules;


import android.support.v4.app.Fragment;
import org.owntracks.android.injection.modules.SupportFragmentModule;
import org.owntracks.android.injection.scopes.PerFragment;
import org.owntracks.android.ui.welcome.version.VersionFragment;

import dagger.Binds;
import dagger.Module;

@Module(includes = SupportFragmentModule.class)
public abstract class VersionFragmentModule {

    @Binds
    @PerFragment
    abstract Fragment bindSupportFragment(VersionFragment f);
}

File: project/app/src/main/java/org/owntracks/android/ui/welcome/version/VersionFragmentModule.kt
package org.owntracks.android.ui.welcome.version

import androidx.fragment.app.Fragment
import dagger.Binds
import dagger.Module
import org.owntracks.android.injection.scopes.PerFragment

@Module
abstract class VersionFragmentModule {
    @Binds
    @PerFragment
    abstract fun bindSupportFragment(f: VersionFragment?): Fragment?
}